tests/Unit/Models/RoomModelTest.php

```diff
@@ -17,5 +17,215 @@ public function create()
     {
         $model = Room::factory()->create();
         $this->assertNotNull($model);
+
+        return $model;
+    }
+
+    /**
+    @test
+    *@depends create
+    */
+    public function it_has_a_location(Room $model)
+    {
+        $this->assertNotNull($model->location_id);
+    }
+
+    /**
+    @test
+    *@depends create
+    */
+    public function it_has_a_theme(Room $model)
+    {
+        $this->assertNotNull($model->escaperoom_theme_id);
+    }
+
+    /**
+    @test
+    *@depends create
+    */
+    public function it_has_a_rate(Room $model)
+    {
+        $this->assertNotNull($model->escaperoom_rate_id);
+    }
+
+    /**
+    @test
+    *@depends create
+    */
+    public function it_has_a_supervision(Room $model)
+    {
+        $this->assertNotNull($model->supervision_id);
+    }
+
+    /**
+    @test
+    *@depends create
+    */
+    public function it_has_a_openAt_date(Room $model)
+    {
+        $this->assertNotNull($model->opened_at);
+    }
+
+    /**
+    @test
+    *@depends create
+    */
+    public function it_has_participants(Room $model)
+    {
+        $this->assertNotNull($model->participants);
+    }
+
+    /**
+    @test
+    *@depends create
+    */
+    public function it_has_participants_private(Room $model)
+    {
+        $this->assertNotNull($model->participants_private);
+    }
+
+    /**
+    @test
+    *@depends create
+    */
+    public function it_has_reset_time(Room $model)
+    {
+        $this->assertNotNull($model->reset_time);
+    }
+
+    /**
+    @test
+    *@depends create
+    */
+    public function it_has_occupied_time(Room $model)
+    {
+        $this->assertNotNull($model->occupied_time);
+    }
+
+    /**
+    @test
+    *@depends create
+    */
+    public function it_has_priority(Room $model)
+    {
+        $this->assertNotNull($model->priority);
+    }
+
+    /**
+    @test
+    */
+    public function it_has_title_attribute()
+    {
+        $location = app('location')::factory()->create();
+        $theme = app('escaperoom_theme')::factory()->create();
+        $room = app('room')::factory()->create(['escaperoom_theme_id' => $theme->id, 'location_id' => $location->id]);
+        $this->assertEquals($room->getTitleAttribute(), $room->theme->title);
```

Comment:
@drewroberts this was working in my first test, but now I got empty when `$room->theme`...... `room` has the relation with `theme`, 

    public function theme()
    {
        return $this->belongsTo(app('escaperoom_theme'));
    }

I'm missing something?

Reply:
I don't know. Looking at the test failures, it seems like it is being passed a `null` value.

Reply:
@pdbreen any clue on this?

Reply:
In tests, you should use the real models (ie, `Location::factory()->create()`).  If the real model isn't defined, it either means a required composer dependency is missing - or, if that dependency is not allowed, there is a need to rework the code to cross package boundaries using an interface.  Always using the real model in tests will help locate these dependency issues faster.

 For relations in factories, its more common to associate the id field with an object - so `['escaperoom_theme_id' => $theme, 'location_id' => $location]` in this case.  Without that, you may need to do a `$room->refresh()` to reload the relations.  
 
Also - the last `assertEquals` is backwards.  The expected value should be the first parameter, the value being tested passed second (make a difference in how failure is reported).  It would also be better to create a theme with a fixed, known title and then test against that constant value.